### PR TITLE
Add test DB for testing on docker

### PR DESCRIPTION
**Pull request by waterkip**
This is a couple of commits that fixes a lot of the testsuite. We still aren't were we want to be, but there is progress.

```
Test Summary Report
-------------------
t/02config.t       (Wstat: 3840 Tests: 196 Failed: 15)
Failed tests:  32-33, 48-49, 80, 82, 84, 91, 95, 98, 108-109
193-194, 196
Non-zero exit status: 15
t/11talk.t         (Wstat: 256 Tests: 22 Failed: 1)
Failed test:  8
Non-zero exit status: 1
t/database.t       (Wstat: 1024 Tests: 9 Failed: 4)
Failed tests:  3, 7-9
Non-zero exit status: 4
t/dispatcher.t     (Wstat: 1024 Tests: 5 Failed: 4)
Failed tests:  2-5
Non-zero exit status: 4
Files=28, Tests=3405, 17 wallclock secs ( 0.33 usr  0.09 sys + 13.25 cusr  1.10 csys = 14.77 CPU)
Result: FAIL
```

### CODE REVIEW

File: t/04i18n.t [hunk 1]
```diff
@@ -1,9 +1,12 @@
...
-
 use strict;
-use Act::Config;
+use warnings;
 
 use Test::More;
+use Test::Lib;
+
+use Act::Config;
+Act::Config->load_configs('t');
```
Comment (HaraldJoerg) [inline t/04i18n.t]: The repository does not have configuration files under t/.  If they are in your working copy and you intended to push them, please review .gitignore which excludes conf and actdocs.

Comment (waterkip) [reply t/04i18n.t]: No, this is a remnant of custom configuration files.

**Comment by HaraldJoerg** [comment]
After some annoying shuffling of directories I got it down to two failing test files:

- `database.t` tests whether a failure to connect to the database or a bad version is reported at startup and not at the first web request which happens at sometime later when no one is looking. I'd say such a test makes perfect sense in a docker environment, too, where application and database reside in independent containers.  The problem is not with the test suite but with lib/Act/Store/Database.pm, where by commit 1ce96acb7 the version check just returns undef.
- `dispatcher.t` tests whether the URLs offered by the web services are routed to the correct files and handlers.  The dispatcher has been changed, so the tests need to be amended accordingly.  It appears that `Dispatcher.pm` needs some care, too, but this is not in scope for this PR.

Related: 89cffdf7 needs to be reverted with the undoing of the changes in I18N.pm.  With a working I18N environment, the <a> element is provided by the .po files and must not be part of the templates.

**Comment by waterkip** [comment]
Can you tell me what you in your act.ini, and what your conference.ini looks like. Because my tests are failing hard on t/02config.t and t/11talk.t? These still fail for me.

File: t/04i18n.t (hunk 1 above)
Comment (waterkip) [reply t/04i18n.t]: And I removed it

**Comment by HaraldJoerg** [comment]
If you just run t/02config.t in verbose mode, it will tell you which variables are missing.  My own setup is pretty customized with four conferences in different languages, among them Russian (for Unicode tests), so 02config.t runs 296 individual tests.  The conference configs are pulled from the Act-Conferences repository without changes.
When I run t/02config.t with etc/act.ini.dist, I get two failures for database_pg_dump and database_dump_file (they are needed for the backup script in bin), so maybe you just need to add dummy values.  I have:
```
pg_dump     = 1
dump_file   = /tmp/dbdump
```
With the same config act.ini.dist, t/11talks.t fails in my setup because it refers to a test database host which I don't have.  It would be helpful to know which of the tests are failing: maybe the database setup or cleanup after the tests fails.  I have no indication why t/11talks.t should behave different from any other tests using the database, It is using a non-existing conference, so the conference setup should not matter.

**Comment by waterkip** [comment]
I already found out. My branch was on v10 for DB upgrades and yours is v12. So I added them in 
7aa58aa. t/02config keeps failing right now on the the payment type

**Comment by waterkip** [comment]
Ok, that one is also fixed.

**Comment by waterkip** [comment]
Exciting times! :+1: 

```
$ docker-compose run --rm act prove -l
Starting act_act-wiki-db_1 ... done
Starting act_act-db_1      ... done
t/00-compile.t ....... ok
t/00test.t ........... ok
t/02act.t ............ ok
t/02config.t ......... ok
t/02pod.t ............ ok
t/03database.t ....... ok
t/04i18n.t ........... ok
t/04template.t ....... ok
t/05util.t ........... ok
t/06country.t ........ ok
t/08track.t .......... ok
t/09dbbool.t ......... ok
t/10user.t ........... ok
t/11talk.t ........... ok
t/13participation.t .. ok
t/14order.t .......... ok
t/15events.t ......... ok
t/16timeslot.t ....... ok
t/17invoice.t ........ ok
t/18abstract.t ....... ok
t/19news.t ........... ok
t/20tags.t ........... ok
t/auth_password.t .... ok
t/database.t ......... ok
t/dispatcher.t ....... Use of uninitialized value in subroutine entry at /opt/act/lib/Act/Dispatcher.pm line 149.
t/dispatcher.t ....... skipped: This one fails
t/form.t ............. ok
t/image.t ............ ok
t/mw_errorpage.t ..... ok
All tests successful.
Files=28, Tests=3291, 21 wallclock secs ( 0.27 usr  0.05 sys + 16.55 cusr  1.27 csys = 18.14 CPU)
Result: PASS
```

**Pull request merged**